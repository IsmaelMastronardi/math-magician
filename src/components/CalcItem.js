import PropTypes from 'prop-types';

function CalcItem(props) {
  const { item, itemClass, calc } = props;
  return (
    <button
      type="button"
      className={itemClass}
      onClick={calc}
    >
      {item}
    </button>
  );
}

CalcItem.propTypes = {
  item: PropTypes.string.isRequired,
  itemClass: PropTypes.string.isRequired,
  calc: PropTypes.func.isRequired,
};

export default CalcItem;
