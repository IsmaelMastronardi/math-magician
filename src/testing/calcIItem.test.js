import { fireEvent, render } from '@testing-library/react';
import CalcItem from '../components/CalcItem';

describe('Renders a individual buttons for the calculator', () => {
  test('render', () => {
    const myFunc = jest.fn();
    const { btn } = render(<CalcItem
      item="+"
      itemClass="class"
      calc={myFunc}
    />);
    expect(btn).toMatchSnapshot();
  });
  test('press a button', () => {
    const myFunc = jest.fn();
    const { getByTestId } = render(<CalcItem
      item="+"
      itemClass="class"
      calc={myFunc}
    />);
    fireEvent.click(getByTestId('buttonTestId'));
    expect(myFunc).toHaveBeenCalled();
  });
});
