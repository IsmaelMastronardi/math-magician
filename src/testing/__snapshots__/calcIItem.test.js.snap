// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a individual buttons for the calculator render 1`] = `undefined`;
